robottelo/hosts.py

```diff
@@ -0,0 +1,361 @@
+import logging
+from urllib.parse import urljoin
+from urllib.parse import urlunsplit
+
+from broker.hosts import Host
+
+from robottelo.config import settings
+from robottelo.constants import DISTRO_RHEL6
+from robottelo.constants import DISTRO_RHEL7
+from robottelo.constants import REPOS
+from robottelo.helpers import install_katello_ca
+from robottelo.helpers import remove_katello_ca
+
+logger = logging.getLogger(__name__)
+
+
+class ContentHostError(Exception):
+    pass
+
+
+class ContentHost(Host):
```

Comment:
I applaud attempt at keeping interface consistent with `VirtualMachine`, but copying common methods doesn't strike me as the best solution.

Could both classes inherit from single base class implementing common operations? Could ContentHost wrap broker vm inside robottelo VirtualMachine? VM is written for current infrastructure, and it definitely shows, but no action is really done until you call `create()`; I think there are some tests that put pre-existing machines inside VM class.

Reply:
The intention behind this version is to make the conversion of existing tests easier. I hope that with the new infrastructure, the VirtualMachine interface will be deprecated, and methods that are no longer useful will be removed from the ContentHost class. For that reason, I don't think we should put effort into a common parent class.
I like where your head is at though!

Reply:
I can see uses for this for CapsuleVirtualMachine, but for lightweight VirtualMachine without much setup, can we incorporate containers instead?